core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -129,6 +119,7 @@ protected void validate(TableMetadata base) {
 
     if (conflictDetectionFilter != null && base.currentSnapshot() != null) {
       validateAddedDataFiles(base, startingSnapshotId, conflictDetectionFilter, caseSensitive);
+      validateAddedDeleteFiles(base, startingSnapshotId, conflictDetectionFilter, caseSensitive);
```

Comment:
Couldn't this use the existing validation that passes a list of data files that were removed? Why not do that?

Reply:
Could you point to the method you refer?

Reply:
I think you mean `validateNoNewDeletesForDataFiles`. We can make that work. Just need to extend it to also accept an optional filter to ignore delete files in other partitions (performance optimization only).

Reply:
We may even keep the old name. We must always validate there are no new deletes for files we read. We can push down the conflict resolution filter if it present but knowing the list of read files should be sufficient.

Thoughts, @rdblue?

Reply:
Basically, keep `validateNoConflictingAppends` as is and always check for new delete files that may match the overwritten data files.

Reply:
Validating concurrent appends is optional (serializable vs snapshot) while validating concurrent deletes is required to avoid corrupting the table.

Reply:
Yeah, I think it would be good to keep the two separate. That would allow fine-grained control of validation and avoid the need to rename the validation config method -- just add a second one. Would we need to validate there are no conflicting deletes in all cases? Seems like we wouldn't necessarily need to if using snapshot isolation.

Reply:
Consider we have a data file A and have a copy-on-write operation that overwrites it with a data file B. If a concurrent operation adds a delete file that references records from the data file A, committing the original overwrite would undelete the rows that were deleted concurrently.

Reply:
This point is important. Let me know if I got you correctly, @rdblue.

Reply:
I think we have discussed it on another thread. Resolving.